modules/packages/LinearAlgebra.chpl

```diff
@@ -1244,49 +1244,38 @@ proc trace(A: [?D] ?eltType) {
 }
 
 private proc _lu (in A: [?Adom] ?eltType) {
-  const n = Adom.shape(0);
-  const LUDom = {1..n, 1..n};
+  const (n,m) = Adom.shape;
 
   // TODO: Reduce memory usage
-  var L, U, LU: [LUDom] eltType;
-
+  var L, LU: [Adom] eltType;
```

Comment:
Previous implementation used to create a local domain `LUDom`, which would cause the result matrix to be a local one, even if the input was distributed.

I think it is fine to change that, however, we need to be aware that this creates a variety of questions as to the performance of this routine. It may also be fine to not answer them at the moment, but at the very least, I would like a comment before this function saying that the performance is not optimized for distributed arrays.

Reply:
Looking at the user-facing `lu`, this change makes me a bit more concerned. I feel like if we are making this change, we should update the doc, saying that the returned matrix has the same distribution as the input matrix.

Reply:
Probably the most important issue with this change that is just occurring to me is 0-based vs 1-based arrays in the LinearAlgebra library. We are working on improving the consistency there, but in the past all the indices in this module was 1-based. And this function shows some remnants of that.

However, the issue with this change here is that, in the previous implementation we were creating a local domain that was 1-based, and using that for `L` and `LU`, but now, we are using ADom which can be anything-based, but the rest of the function is not adjusted to accommodate that.

Easy solution: Revert this part back to declaring a 1-based local domain.
Ideal solution: Adjust the rest of the logic to steer away from 1-based implementation and use `ADom`'s shape.